README.md

```diff
@@ -1,3 +1,137 @@
+
+_DISCLAIMER: This is experimental software. **Use it at your own risk**!_
+
+The solidity library has been designed aiming at decreasing gas consumption and its complexity due to EC operations.
+
+It provides two main `pure` functions for verifying VRF proofs:
+
+- **verify**:
+  - _Description_: VRF *full* verification (requires heavy EC computation)
+  - _Inputs_:
+    - *_publicKey*: The public key as an array composed of `[pubKey-x, pubKey-y]`
+    - *_proof*: The VRF proof as an array composed of `[gamma-x, gamma-y, c, s]`
+    - *_message*: The message (in bytes) used for computing the VRF
+  - _Output_:
+    - true, if VRF proof is valid
+- **fastVerify** (low gas consumption): VRF *fast* verification by providing additional EC points. It uses the `ecrecover` precompiled function to verify EC multiplications.
+  - _Inputs_:
+    - *_publicKey*: The public key as an array composed of `[pubKey-x, pubKey-y]`
+    - *_proof*: The VRF proof as an array composed of `[gamma-x, gamma-y, c, s]`
+    - *_message*: The message (in bytes) used for computing the VRF
+    - *_uPoint*: The `u` EC point defined as `U = s*B - c*Y`
+    - *_vComponents*: The components required to compute `v` as `V = s*H - c*Gamma`
+  - _Output_:
+    - true, if VRF proof is valid
+
+Additionally, the library provides some auxiliary `pure` functions to facilitate computing the aforementioned input parameters:
+
+- **decodeProof**:
+  - _Description_: Decode from bytes to VRF proof
+  - _Input_:
+    - *_proof*: The VRF proof as an array composed of `[gamma-x, gamma-y, c, s]`
+  - _Output_:
+    - The VRF proof as an array composed of `[gamma-x, gamma-y, c, s]`
+- **decodePoint**:
+  - _Description_: Decode from bytes to EC point
+  - _Input_:
+    - *_point*: The EC point as bytes
+  - _Output_:
+    - The point as `[point-x, point-y]`
+- **computeFastVerifyParams**:
+  - _Description_: Compute the parameters (EC points) required for the VRF fast verification function
+  - _Inputs_:
+    - *_publicKey*: The public key as an array composed of `[pubKey-x, pubKey-y]`
+    - *_proof*: The VRF proof as an array composed of `[gamma-x, gamma-y, c, s]`
+    - *_message*: The message (in bytes) used for computing the VRF
+  - _Output_:
+    - The fast verify required parameters as the tuple `([uPointX, uPointY], [sHX, sHY, cGammaX, cGammaY])`
+
+## Elliptic Curve VRF (using `Secp256k1`)
+
+This library follows the algorithms described in [VRF-draft-04](https://tools.ietf.org/pdf/draft-irtf-cfrg-vrf-04) in order to provide the VRF verification capability.
+
+The supported cipher suite is `SECP256K1_SHA256_TAI`, i.e. the aforementioned algorithms using `SHA256` as digest function and the `secp256k1` curve.
+
+For elliptic curve arithmetic operations `vrf-solidity` uses the `elliptic-curve-solidity` library.
+
```

Comment:
Shall we mention the fact that the SECP256K1_SHA256_TAI IS 0XFF ciphersuite?

Reply:
I included the `0xFE` cipher suite.